src/app/context.js

```diff
@@ -301,41 +301,56 @@ export default class Context {
      * @memberof Context
      */
     setUpLuts() {
-        this.luts_png.url =
-            this.server + this.getPrefixedURI(WEBGATEWAY, true) + LUTS_PNG_URL;
-
-        // determine the luts png height
-        let lutsPng = new Image();
-        lutsPng.onload = (e) => {
-            this.luts_png.height = e.target.naturalHeight;
-            this.luts_png.image = lutsPng;
-            for (let [id, conf] of this.image_configs) conf.changed();
-        }
-        lutsPng.src = this.luts_png.url;
-
-        // now query the luts list
-        let server = this.server;
-        let uri_prefix =  this.getPrefixedURI(WEBGATEWAY);
         $.ajax(
-            {url : server + uri_prefix + "/luts/",
+            {url : this.server + this.getPrefixedURI(IVIEWER) + "/is_dynamic_lut/",
```

Comment:
I'm wondering if you could save yourself a call (and not need the new `is_dynamic_lut/` endpoint) if you simply load the `/luts/` json as before, then check for `response.png_luts_new` and if this exists then `is_dynamic_lut` is True?

I just tried this and my diff looks like this (not updated the indents yet to minimise the diff):

```
+++ b/src/app/context.js
@@ -302,11 +302,12 @@ export default class Context {
      */
     setUpLuts() {
         $.ajax(
-            {url : this.server + this.getPrefixedURI(IVIEWER) + "/is_dynamic_lut/",
+            {url : this.server + this.getPrefixedURI(WEBGATEWAY) + "/luts/",
             success : (response) => {
                 // Check first whether omero-web can provides LUT dynamically
                 // and set URL accordingly
-                let is_dynamic_lut = response.is_dynamic_lut;
+                let is_dynamic_lut = Boolean(response.png_luts_new);
                 if (is_dynamic_lut) {
                     this.luts_png.url =
                         this.server + this.getPrefixedURI(WEBGATEWAY, false) + "/luts_png/";
@@ -324,15 +325,6 @@ export default class Context {
                 }
                 lutsPng.src = this.luts_png.url;
 
-                // now query the luts list
-                let server = this.server;
-                let uri_prefix =  this.getPrefixedURI(WEBGATEWAY);
-                $.ajax(
-                    {url : server + uri_prefix + "/luts/",
-                    success : (response) => {
-                        if (typeof response !== 'object' || response === null ||
-                            !Misc.isArray(response.luts)) return;
-
                         if (is_dynamic_lut) {
                             // If it's dynamic, uses the new list instead
                             response.png_luts = response.png_luts_new
@@ -349,8 +341,6 @@ export default class Context {
                                 this.luts.set(mapValue.name, mapValue);
                             });
                         for (let [id, conf] of this.image_configs) conf.changed();
-                    }
-                });
             }
         });
     }
```

It seems to work for me. Does that make sense?

Reply:
It makes sense yes. I think the reason I did that is because I originally returned a single LUT list with the updated `this.getPrefixedURI(WEBGATEWAY) + "/luts/"`

But since it now returns `png_luts_new` in the response, there's a way to know if it's the old or new response.

Can you please commit these changes?

There should be a similar solution for omero-figure then. I'll look into it.